lib/ansible/modules/cloud/azure/azure_rm_monitordiagnosticsettings_facts.py

```diff
@@ -0,0 +1,259 @@
+            type: str
+            example: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroup/myResourceGroup/providers/
+                      microsoft.network/publicipaddresses/myip/providers/microsoft.insights/diagnosticSettings/myipdiagnostic"
+        storage_account_id:
+            description:
+                - The resource ID of the storage account to which you would like to send Diagnostic Logs.
+            returned: always
+            type: str
+            sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/myorgdiag"
+        name:
+            description:
+                - The name of the resource.
+            returned: always
+            type: str
+            sample: myipdiagnostic
+        logs:
+            description:
+                - The list of logs settings.
+            returned: always
+            type: list
+            sample: '[
+                        {
+                            "category": "DDoSProtectionNotifications",
+                            "enabled": false,
+                            "retention_policy": {
+                                "days": 0,
+                                "enabled": false
+                            }
+                        },
+                        {
+                            "category": "DDoSMitigationFlowLogs",
+                            "enabled": true,
+                            "retention_policy": {
+                                "days": 6,
+                                "enabled": true
+                            }
+                        },
+                        {
+                            "category": "DDoSMitigationReports",
+                            "enabled": true,
+                            "retention_policy": {
+                                "days": 3,
+                                "enabled": true
+                            }
+                        }
+                    ]'
+        metrics:
+            description:
+                - The list of metric settings.
+            returned: always
+            type: list
+            sample: '[
+                        {
+                            "category": "AllMetrics",
+                            "enabled": false,
+                            "retention_policy": {
+                                "days": 0,
+                                "enabled": false
+                            }
+                        }
+                    ]'
+category:
+    description: A list of dictionaries containing facts for diagnostic setting category.
+    returned: when I(show_category) is set to true
+    type: complex
+    contains:
+        category_type:
+            description:
+                - The type of the diagnostic settings category. Possible values include 'Metrics' and 'Logs'
+            returned: always
+            type: str
+            example: "Logs"
```

Comment:
what user should pass in in main module for `category`?  `logs`?

Reply:
For the type of the category, two possible values are Logs and Metrics. In the main module, two parameters named ```logs``` and ``` metrics``` are available for user to set. It's consistent with the ```az cli```.

Reply:
Actually, I think the word ```category``` is a little bit far from ```diagnostic settings```. The word ```metrics``` and ```logs``` are more meaningful for user in the main module.